### DISCUSSION TITLE
通常比较复杂的错误处理一般会怎么做？有什么第三方框架？

### DISCUSSION THREAD

**Discussion by hhfwork** [BYJRK/DotNet-Discussions#317] [C# 技术交流]
目前在工作中用wpf加IHost，对于错误处理只会用Try-Catch-Finally，感觉随着代码的增多会越来越复杂，而且异步多线程加上会更繁琐，而且有些是用户错误，有些是自身程序错误，还有下位机上抛错误，不知道大神们平时是怎么处理的，还是会去使用第三方库？亦或者用[IExceptionHandler](https://learn.microsoft.com/zh-cn/aspnet/core/fundamentals/error-handling?view=aspnetcore-8.0), asp的管道中间件，aop，函数式？上面的这些我问chartgpt的，感觉有共性，但是没实践过。希望给些建议或方向？

**Comment by BYJRK** [comment]
一般来说，下位机可以自己写一个 Exception，里面还可以包含自定义的错误码等信息。异常的处理可以用抛异常并在上层捕获，然后看情况进行处理，比如包装为更上层的 Exception;继续抛出，或打日志等。

**Reply by hhfwork** [reply]
那有什么更好的管理异常的工具吗？现在一个函数里往往代码没几句，光异常处理就2、30行下去了，而且随着代码切分越多需要抛出的异常也会越多，管理起来很麻烦。
感觉就像处理之前没用CommunityToolkit.Mvvm的时候每次都要写OnPropertyChanged的一样繁琐。

**Reply by CodingOctocat** [reply]
真会有那么多异常吗？发个函数看看。

**Reply by CodingOctocat** [reply]
那你试试 Polly？

**Reply by hhfwork** [reply]
Polly感觉有点太重了，时间所迫后面我还是根据寒流大佬的思路来了，现在在学习原生成器，这个应该就可以通过特性来添加外包函数和生成新的函数专属的exception了，就像ObservableProperty一样

**Reply by CodingOctocat** [reply]
用 [MethodBoundaryAspect.Fody](https://github.com/vescon/MethodBoundaryAspect.Fody) 会不会更简单

**Comment by hhfwork** [comment]
算是搞定了
